A massive commit containing many theme enhancements, bugfixes and a coup...

matthew-kerle opened the pull request:
...le of additions.

!enhancement - News fader now only "fades" if there are multiple items.
({js_dir}fader.js)
!enhancement - Search button vertically aligned.
({theme_dir}index.template.php, {css_dir}index.css)
!enhancement - Mark all messages as read alignment fix.
({css_dir}index.css)
!bugfix - When opt.bDisabled is false for the JumpTo JavaScript object,
there is a "0" right before the select name. <select 0name="whatever".
This is caused by: <select ' + (this.opt.bDisabled == true ?
'disabled="disabled" ' : 0) the else should be an empty string, rather
than an integer so nothing outputs to the markup.
!enhancement - CSS "rfix" class. This class should be utilized whenever
you are 'integrating' a header (cat_bar / title_bar) with a roundframe.
This class gets rid of the one-pixel space between the header and the
body that looks like crap. ({css_dir}index.css)
!enhancement - Admin welcome header margin resolved by removing margin
in CSS (admin.css), and adding "rfix" class to the roundframe element
in the actual template. ({theme_dir}admin.template.php,
{css_dir}admin.css)
!enhancement - Admin quick search CSS vertical-alignment fixes.
!enhancement - Posting Icons vertically aligned properly.
!enhancement - ?action=who has an inconsistency. The second page index
and "show" select box are outside the main content box, yet the top one
is inside of it. Moved the second one back into the main box and fixed
all alignment issues with the second box.
!enhancement - ?action=who the upper show box had some inconstancies
with alignment in comparison to the pages block.
!enhancement - Alignment fix with the unapprove image sprite on posts
"more" drop down.
!bugfix - The CSS "divider" class for dividing boards on the BoardIndex
was set to: .table_list .divider so it had no affect, it was designed
to add a divider between boards. It should be applied to: .table_list
.divider tr td.
!enhancement - Fixed alignment of unread post icons and mark all
messages as read. Why this issue was fixed twice is because the "first"
fix did not incorporate room for the divider which was broken.
!enhancement - Fixed board icon vertical align.
!bugfix - BoardIndex layout was off with headers and their board
content. This was a big BoardIndex layout.
!enhancement - Stats block of boards borders are now #ddd; as opposed
to #ccc which is not the colour of the other borders.
!bugfix - Undefined Index "rss_limit" in index.template.php. Simply
added && !empty($modSettings['rss_limit']) for the condition of the RSS
link being shown, since the variable is used if it's true, it needs
substance.
!enhancement - Added "short_description" for boards, and utilized
"short_description" for the title="" of child boards that are redirects
so that when you mouseover them, you also see a "short" description of
the redirect without having to enter the parent. I set the length on
shorten_subject(); to 128 for now, but can definitely be lowered later
if we feel necessary.
!bugfix - Added "Members Viewing Topic" block to topics, since it was
left out of topics.
(http://dev.simplemachines.org/mantis/view.php?id=5004)
!bugfix - Code blocks have a minimum height of 10em, which is too
large. By removing the height attribute in the CSS it allows the block
to shrink or expand based on how much is inputted.
(http://dev.simplemachines.org/mantis/view.php?id=5003)

{theme_dir} = ./Themes/default/
{css_dir} = ./Themes/default/css/
{js_dir} = ./Themes/default/scripts

Commit:
A massive commit containing many theme enhancements, bugfixes and a couple of additions.

!enhancement - News fader now only "fades" if there are multiple items.
({js_dir}fader.js)
!enhancement - Search button vertically aligned.
({theme_dir}index.template.php, {css_dir}index.css)
!enhancement - Mark all messages as read alignment fix.
({css_dir}index.css)
!bugfix - When opt.bDisabled is false for the JumpTo JavaScript object,
there is a "0" right before the select name. <select 0name="whatever".
This is caused by: <select ' + (this.opt.bDisabled == true ?
'disabled="disabled" ' : 0) the else should be an empty string, rather
than an integer so nothing outputs to the markup.
!enhancement - CSS "rfix" class. This class should be utilized whenever
you are 'integrating' a header (cat_bar / title_bar) with a roundframe.
This class gets rid of the one-pixel space between the header and the
body that looks like crap. ({css_dir}index.css)
!enhancement - Admin welcome header margin resolved by removing margin
in CSS (admin.css), and adding "rfix" class to the roundframe element
in the actual template. ({theme_dir}admin.template.php,
{css_dir}admin.css)
!enhancement - Admin quick search CSS vertical-alignment fixes.
!enhancement - Posting Icons vertically aligned properly.
!enhancement - ?action=who has an inconsistency. The second page index
and "show" select box are outside the main content box, yet the top one
is inside of it. Moved the second one back into the main box and fixed
all alignment issues with the second box.
!enhancement - ?action=who the upper show box had some inconstancies
with alignment in comparison to the pages block.
!enhancement - Alignment fix with the unapprove image sprite on posts
"more" drop down.
!bugfix - The CSS "divider" class for dividing boards on the BoardIndex
was set to: .table_list .divider so it had no affect, it was designed
to add a divider between boards. It should be applied to: .table_list
.divider tr td.
!enhancement - Fixed alignment of unread post icons and mark all
messages as read. Why this issue was fixed twice is because the "first"
fix did not incorporate room for the divider which was broken.
!enhancement - Fixed board icon vertical align.
!bugfix - BoardIndex layout was off with headers and their board
content. This was a big BoardIndex layout.
!enhancement - Stats block of boards borders are now #ddd; as opposed
to #ccc which is not the colour of the other borders.
!bugfix - Undefined Index "rss_limit" in index.template.php. Simply
added && !empty($modSettings['rss_limit']) for the condition of the RSS
link being shown, since the variable is used if it's true, it needs
substance.
!enhancement - Added "short_description" for boards, and utilized
"short_description" for the title="" of child boards that are redirects
so that when you mouseover them, you also see a "short" description of
the redirect without having to enter the parent. I set the length on
shorten_subject(); to 128 for now, but can definitely be lowered later
if we feel necessary.
!bugfix - Added "Members Viewing Topic" block to topics, since it was
left out of topics.
(http://dev.simplemachines.org/mantis/view.php?id=5004)
!bugfix - Code blocks have a minimum height of 10em, which is too
large. By removing the height attribute in the CSS it allows the block
to shrink or expand based on how much is inputted.
(http://dev.simplemachines.org/mantis/view.php?id=5003)

{theme_dir} = ./Themes/default/
{css_dir} = ./Themes/default/css/
{js_dir} = ./Themes/default/scripts

emanuele45 commented on Sources/Subs-BoardIndex.php:
```diff
@@ -172,6 +172,7 @@ function getBoardIndex($boardIndexOptions)
 				'id' => $row_board['id_board'],
 				'name' => $row_board['board_name'],
 				'description' => $row_board['description'],
+				'short_description' => shorten_subject($row_board['description'], 128),
```
/me feels that needs strip_tags or something

Commit:
included strip_tags(); for short_description since it's used in a title.

What was I thinking? I guess I rarely put things into the title
attribute. There we go. Tested.

matthew-kerle commented on Sources/Subs-BoardIndex.php:
```diff
@@ -172,6 +172,7 @@ function getBoardIndex($boardIndexOptions)
 				'id' => $row_board['id_board'],
 				'name' => $row_board['board_name'],
 				'description' => $row_board['description'],
+				'short_description' => shorten_subject(strip_tags($row_board['description']), 128),
```
Sorry about that. Rarely put things into the title attribute I guess. All better now and tested. :+1:

Spuds commented:
This is a PR to the development branch and looks to be based on that branch, not the 2.1 branch .... That development branch is ~230 commits behind the current 2.1 branch ... so some of these errors do exist in the 2.1 branch for sure, some I just don't know.  The PR will need to be redone for the 2.1 branch ... welcome to github, bout time you got your arse here :P

matthew-kerle commented:
Gosh darnit...so basically I spent hours writing fixes for things that no longer exist? Crud.

I've actually been around GitHub for quite some time, although not been active at all until today.

How should I work on merging what I did with the active branch? (where is the active branch, too?)

emanuele45 commented:
/me feels it's time to update development with release-2.1 things...

emanuele45 commented:
Well, TBH at that point I really wonder if it (still) make sense to keep the release-2.1 branch...

Github master @Norv ?
What do you think?

Spuds commented:
Well I know for sure that some of the bugs you found are still in the 2.1 branch so no lost time there .. as for the theme updates I can't really say ... there was a large theme cleanup commit via Ant but that does not mean everything was fixed, far from it really.

norv commented:
@emanuele45 , how would you add fixes to 2.1 in parallel with [not for next release] functionality, if everything would be in /development?
I understand that we're aiming at nvie model, and it makes sense, if a little 'busy' with 3 main branch types.

I'd say continue with the model, but you need to change the default branch to release-2.1 (so that people would see, checkout and PR to that branch, first), merge in development the current state, and tbh it'd be best to revert master to 2.0 (2.0.2) state. Master should always be the latest stable state.

I think one of the reasons why it seems a little much atm (a lot to keep merging in development) is that for this project, we have (and will have) lots of changes and updates to release-x.x after branching off development...

emanuele45 commented:
Well, my reasoning is/was more along the lines of:
1. at the moment we have a "development" branch where nothing is happening,
2. sooner or later fixes in the 2.1 branch should be merged into development,
3. many things have been added after the branching.
   In other words: probably the creation of the release-2.1 branch was a bit too rushed.

In the long run I believe the model is fine, just not sure **now** it is meaningful. ;)

norv commented:
**Side note**: _how would you add fixes to 2.1 in parallel with [not for next release] functionality_ 
I asked this question because of two reasons: as a 'catch-all' for any useful PRs (that don't belong to 2.1, and it remains to be seen where they may belong), and, because I've seen you folks talking about further development on the current codebase (aka 2.2).
Please don't take the above as endorsement from me of further development ("2.2") on the current codebase, however. IMO, any further development has to make major changes. That's a major version. Just to be clear on this, FWIW.

**On the last point:** Ema, yes, 3. is the problem. I think release-2.1 was created when all known, major, and necessary functionality for 2.1 was in. That makes sense (trying to feature-freeze). But then, the issue is,
- Theory says that from there, there are 'only some bug fixes for it'.

_vs_ 
- Practice on SMF 2.0+ codebase says there's much more work and rework to do than that, until a stable and decent release is out.

That's why it doesn't make much sense atm. I think the release-[minor] branches in the nvie model are meant for a codebase where minor releases are (and can be) really _minor_. This one is a hybrid. :) (as expected, no matter how much you try)
Applied to 2.1, well it kinda means that it'll make more sense at RC1 stage or so, if any. :)

**Back on topic.** I believe the development branch hasn't diverged yet (it's just behind), but I think I've seen some PRs meant for it though, already. Is my memory failing me again? (not surprising if it does, I blame age :P).
